server/services/interfaces/locationDetails.ts

```diff
@@ -0,0 +1,20 @@
+import { StaffDetails } from '../../interfaces/prisonApi/staffDetails'
+
+export interface LocationDetails {
+  agencyId: string
+  agencyName: string
+  livingUnitId: number
+  location: string
+  isTemporaryLocation: boolean
+  assignmentDateTime: string
+  assignmentEndDateTime: string
+  movementMadeByUsername: string
+  movementMadeByStaffDetails?: StaffDetails
+}
+
+export interface LocationDetailsGroupedByPeriodAtAgency {
+  agencyName: string
+  fromDate: string
+  toDate: string
+  locationDetails: LocationDetails[]
+}
```

Comment:
Service layer specific interface introduced without formatting of dates and names etc.